ios/Capacitor/Capacitor/WebViewAssetHandler.swift

```diff
@@ -181,22 +187,32 @@ open class WebViewAssetHandler: NSObject, WKURLSchemeHandler {
                             httpVersion: nil,
                             headerFields: mergedHeaders
                         ) {
-                            urlSchemeTask.didReceive(modifiedResponse)
+                            self.lock(urlSchemeTask, { urlSchemeTask.didReceive(modifiedResponse)})
                         }
                     }
 
                     if let data = data {
-                        urlSchemeTask.didReceive(data)
+                        self.lock(urlSchemeTask, { urlSchemeTask.didReceive(data)})
                     }
                 }
             }
-            urlSchemeTask.didFinish()
+            self.lock(urlSchemeTask, { urlSchemeTask.didFinish() })
+            self.stoppedTasks.remove(urlSchemeTask.hash)
             return
         }
 
         task.resume()
     }
 
+    private func lock(_ urlSchemeTask: WKURLSchemeTask, _ action: @escaping () -> Void) {
+        let group = DispatchGroup()
+        group.enter()
+        if !self.stoppedTasks.contains(urlSchemeTask.hash) {
+            action()
+        }
+        group.leave()
+    }
```

Comment:
This use of DispatchGroup isn't doing anything. You would typically use DispatchGroup in tandem with a DispatchQueue, and then you would do `group.wait()` to make sure all tasks finished executing before continuing.